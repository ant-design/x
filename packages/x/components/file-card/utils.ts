export function getSize(size: number) {
  let retSize = size;
  const units = ['B', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB'];
  let unitIndex = 0;

  while (retSize >= 1024 && unitIndex < units.length - 1) {
    retSize /= 1024;
    unitIndex++;
  }

  return `${retSize.toFixed(0)} ${units[unitIndex]}`;
}

export function matchExt(suffix: string, ext: string[]) {
  return ext.some((e) => suffix.toLowerCase() === `.${e}`);
}
