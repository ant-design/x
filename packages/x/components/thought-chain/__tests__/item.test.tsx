import React from 'react';
import mountTest from '../../../tests/shared/mountTest';
import rtlTest from '../../../tests/shared/rtlTest';
import themeTest from '../../../tests/shared/themeTest';
import { render } from '../../../tests/utils';
import ThoughtChain from '..';

describe('ThoughtChain.Item Component', () => {
  mountTest(() => <ThoughtChain.Item />);

  rtlTest(() => <ThoughtChain.Item title="ThoughtChainItem.1" />);

  themeTest(() => <ThoughtChain.Item title="ThoughtChainItem.1" />);

  beforeAll(() => {
    jest.useFakeTimers();
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  it('ThoughtChain component work', () => {
    const { container, getByText } = render(<ThoughtChain.Item title="ThoughtChain.Item.title" />);
    const element = container.querySelector<HTMLUListElement>('.ant-thought-chain');
    expect(getByText('ThoughtChain.Item.title')).toBeInTheDocument();
    expect(element).toBeTruthy();
    expect(element).toMatchSnapshot();
  });

  it('ThoughtChain.Item supports ref', () => {
    const ref = React.createRef<any>();
    render(<ThoughtChain.Item ref={ref} title="ref test" />);
    expect(ref.current).not.toBeNull();
  });
});
