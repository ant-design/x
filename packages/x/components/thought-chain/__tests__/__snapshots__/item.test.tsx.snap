// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ThoughtChain.Item Component ThoughtChain component work 1`] = `
<div
  class="ant-thought-chain ant-thought-chain-item ant-thought-chain-item-solid"
>
  <div
    class="ant-thought-chain-item-content"
  >
    <div
      class="ant-thought-chain-item-title"
    >
      ThoughtChain.Item.title
    </div>
  </div>
</div>
`;

exports[`ThoughtChain.Item Component rtl render component should be rendered correctly in RTL direction 1`] = `
<div
  class="ant-thought-chain ant-thought-chain-item ant-thought-chain-item-solid ant-thought-chain-item-rtl"
>
  <div
    class="ant-thought-chain-item-content"
  >
    <div
      class="ant-thought-chain-item-title"
    >
      ThoughtChainItem.1
    </div>
  </div>
</div>
`;

exports[`ThoughtChain.Item Component test theme component should be rendered correctly when configuring the theme.components 1`] = `
<div
  class="ant-thought-chain css-var-«r3» ant-thought-chain-item ant-thought-chain-item-solid"
>
  <div
    class="ant-thought-chain-item-content"
  >
    <div
      class="ant-thought-chain-item-title"
    >
      ThoughtChainItem.1
    </div>
  </div>
</div>
`;
